src/components/search-toolbox.jsx

```diff
@@ -0,0 +1,114 @@
+import React from "react";
+import Button from "@material-ui/core/Button";
+import Radio from "@material-ui/core/Radio";
+import green from "@material-ui/core/colors/green";
+import { withStyles } from "@material-ui/core/styles";
+import TextField from "@material-ui/core/TextField";
+import Grid from "@material-ui/core/Grid";
+import RadioGroup from "@material-ui/core/RadioGroup";
+import FormControlLabel from "@material-ui/core/FormControlLabel";
+import FormLabel from "@material-ui/core/FormLabel";
+const styles = theme => ({
+  root: {
+    minHeight: "30px",
+    margin: "20px",
+    padding: "20px",
+    border: "1px solid grey",
+  },
+  formControl: {
+    margin: theme.spacing.unit * 3
+  },
+  group: {
+    margin: `${theme.spacing.unit}px 0`,
+    flexDirection: "row"
+  },
+  radio: {
+    color: green[600],
+    "&$checked": {
+      color: green[500]
+    }
+  },
+  checked: {},
+  textField: {
+    marginLeft: theme.spacing.unit,
+    marginRight: theme.spacing.unit,
+    width: "100%",
+  },
+  search: {
+    textAlign: "right",
+    alignContent: "flex-end"
+  }
+});
+
+class SearchToolbox extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { query: "", searchBy: this.props.searchBy };
```

Comment:
Note, that you're getting `searchBy` property from props only when component initializes.

If your component will get new props after initialization, it will not reflect the change.
This is a common source of bugs in React apps. A general recommendation is to avoid such props-to-state synchronization wherever possible. If it's not possible to avoid, you should remember to handle prop changes by leveraging [`getDerivedStateFromProps`](https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops).

Reply:
Updated. Please review.